### DISCUSSION TITLE
On the chart editing screen, the icon of dataset should be similar to those used on the “Dataset” screen.

### DISCUSSION THREAD

**Discussion by xavier-GitHub76** [apache/superset#33366] [Ideas]
Hello,

Physical and virtual data sets do not have the same pictogram.
![image](https://github.com/user-attachments/assets/7773669d-ef86-4837-a272-11061eadfb97)
![image](https://github.com/user-attachments/assets/94f2e1c8-fe7c-462d-83c1-61b871335c73)

On the chart editing screen, the icon should be similar to those used on the “Dataset” screen.

![image](https://github.com/user-attachments/assets/ccd73fe0-4115-4904-ab27-6abbc5f23c4f)

**Comment by kasiazjc** [comment]
> Hello,
> 
> Physical and virtual data sets do not have the same pictogram. ![image](https://private-user-images.githubusercontent.com/82046143/352522859-7773669d-ef86-4837-a272-11061eadfb97.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MjIyNTI1NjUsIm5iZiI6MTcyMjI1MjI2NSwicGF0aCI6Ii84MjA0NjE0My8zNTI1MjI4NTktNzc3MzY2OWQtZWY4Ni00ODM3LWEyNzItMTEwNjFlYWRmYjk3LnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDA3MjklMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwNzI5VDExMjQyNVomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPWRlYmI1MDcxZmVlYjdkOTgxZTljYjM3Yzk3MTQ0NTAyNjYwYjE5NmNjMjZkMGI3ZjY2OGJhOTczNzkzNzU3ZWEmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.UmPWZTmrFHuv4iZm-U2D9W5ti0nvfVy_bSTsuWhjJ3c) ![image](https://private-user-images.githubusercontent.com/82046143/352522698-94f2e1c8-fe7c-462d-83c1-61b871335c73.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MjIyNTI1NjUsIm5iZiI6MTcyMjI1MjI2NSwicGF0aCI6Ii84MjA0NjE0My8zNTI1MjI2OTgtOTRmMmUxYzgtZmU3Yy00NjJkLTgzYzEtNjFiODcxMzM1YzczLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDA3MjklMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwNzI5VDExMjQyNVomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTJmOGNiYTMwNzM5ZjA1MjNkMTRkYjFlMzcyYjQ5MzNiY2YwNjI0NGQwYzY0ZmJkNDljOWIyZmUwMjcxZTE5YzUmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.25RX7GqhzB3VAePiS1EE6b8xWT9s7iabGkSssu52IZQ)
> 
> On the chart editing screen, the icon should be similar to those used on the “Dataset” screen.
> 
> ![image](https://private-user-images.githubusercontent.com/82046143/352523184-ccd73fe0-4115-4904-ab27-6abbc5f23c4f.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MjIyNTI1NjUsIm5iZiI6MTcyMjI1MjI2NSwicGF0aCI6Ii84MjA0NjE0My8zNTI1MjMxODQtY2NkNzNmZTAtNDExNS00OTA0LWFiMjctNmFiYmM1ZjIzYzRmLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDA3MjklMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwNzI5VDExMjQyNVomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTE3NTBlYjQyYmQ3YmQ4ZGZhMDkyNGRjOGEyZTY0MGIwZGRkMTExMTNkOGI3Y2E5OWIyZWNkZGFhMTZkY2ZkMGEmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.E6VeyVubCw0UM2UQOCdBIATTnyX4OuM49O1bLpO_Z9w)

Hi, thanks for raising this! I definitely agree, do you think it's something you would be able to change? 🙏

**Comment by xavier-GitHub76** [comment]
I can try but I'm a novice.
Could you help me to identify files to changed ?

After, I will try to see how to add the control on the kind of dataset and apply the good SVG : 

Physical dataset :
```
<svg xmlns="http://www.w3.org/2000/svg" width="1em" height="1em" fill="currentColor" aria-hidden="true" focusable="false" class="" viewBox="0 0 24 24">
    <path fill="currentColor" fill-rule="evenodd" d="M21 2.5H3a1 1 0 0 0-1 1v18a1 1 0 0 0 1 1h18a1 1 0 0 0 1-1v-18a1 1 0 0 0-1-1m-13 18H4v-4h4zm0-6H4v-4h4zm0-6H4v-4h4zm6 12h-4v-4h4zm0-6h-4v-4h4zm0-6h-4v-4h4zm6 12h-4v-4h4zm0-6h-4v-4h4zm0-6h-4v-4h4z" clip-rule="evenodd"></path>
</svg>
```

Virtual Dataset :
```
<svg xmlns="http://www.w3.org/2000/svg" width="1em" height="1em" fill="currentColor" aria-hidden="true" focusable="false" class="" viewBox="0 0 24 24"> 
    <path fill="#136478" d="M4 4.5h16v16H4z"></path>
    <path fill="#A5DBE9" fill-rule="evenodd" d="M21 2.5H3a1 1 0 0 0-1 1v18a1 1 0 0 0 1 1h18a1 1 0 0 0 1-1v-18a1 1 0 0 0-1-1m-13 18H4v-4h4zm0-6H4v-4h4zm0-6H4v-4h4zm6 12h-4v-4h4zm0-6h-4v-4h4zm0-6h-4v-4h4zm6 12h-4v-4h4zm0-6h-4v-4h4zm0-6h-4v-4h4z" clip-rule="evenodd"></path>
</svg>

```

**Comment by qingshan007** [comment]
**Problem Investigation:**

After investigation, it was found that the rendering type was incorrect
Create a chart type that is always table
datasource type: table

On the chart construction page, switch datasets (virtual or physical):

datasource type: virtual
datasource type: physical

Without modifying or destroying the original logic of the newly created dataset:
Locate to: superset-frontend\src\explore\components\controls\DatasourceControl\index.jsx

1. Add judgment logic, where the physical dataset SQL field is always='', and the virtual one is the SQL during construction:
````
const dynamicDatasourceType = datasource?.sql
      ? DatasourceType.Query
      : DatasourceType.Table;
````

2. Use a new judgment:
```
{/* {datasourceIconLookup[datasource?.type]} */}
↓
{datasourceIconLookup[dynamicDatasourceType]}
```

3. Modify the icon:

```
// Assign icon for each DatasourceType.   If no icon assingment is found in the lookup, no icon will render
// export const datasourceIconLookup = {
//   [DatasourceType.Query]: (
//     <Icons. ConsoleSqlOutlined className="datasource-svg" />
//   ),
//   [DatasourceType.Table]: <Icons.DatasetPhysical className="datasource-svg" />,
// };
↓
export const datasourceIconLookup = {
[DatasourceType.Query]: <Icons.DatasetVirtual className="datasource-svg" />,
[DatasourceType.Table]: <Icons.DatasetPhysical className="datasource-svg" />,
};
```

4. Result:
Switch datasets and display icons of corresponding types
Create a regular chart OR by entering it from the dataset page and displaying the corresponding type of icon

**Comment by qingshan007** [comment]
@xavier-GitHub76 @kasiazjc

**Comment by rusackas** [comment]
I agree with this... and I would encourage anyone to open a PR to this effect. Meanwhile, since it's not a bug, but rather an enhancement (and it's been silent for a while), I'll convert this to an Ideas Discussion where it can live on in hopes of finding a volunteer :D

**Comment by xavier-GitHub76** [comment]
The PR https://github.com/apache/superset/pull/33977 is open

**Comment by msyavuz** [comment]
Thanks a lot @xavier-GitHub76 for noticing the difference and fixing it
